hydra-node/test/Hydra/BehaviorSpec.hs

```diff
@@ -606,10 +605,11 @@ simulatedChainAndNetwork initialChainState = do
   -- seconds
   blockTime = 20
 
-  simulateTicks nodes = forever $ do
+  simulateTicks nodes chainStateVar = forever $ do
+    cs <- readTVarIO chainStateVar
     threadDelay blockTime
     now <- getCurrentTime
-    readTVarIO nodes >>= \ns -> mapM_ (`handleChainEvent` Tick now) ns
+    readTVarIO nodes >>= \ns -> mapM_ (`handleChainEvent` Tick now (chainStateSlot cs)) ns
```

Comment:
This is smelly. Don't read a TVar (`chainStateVar` here) and then wait.. it might have changed until you use this value (unless you deliberately want to take the value before waiting, but I doubt this is what we want here).

In general, it's always a good idea to group multiple STM actions into a single `atomically` block to ensure things are consistent, especially as we are writing to `chainStateVar` from other threads!

Must: read chain state just before calling back with that slot